AUTHORING_MANIFESTS.md

```diff
@@ -128,8 +128,10 @@ If composed solely of numbers and dots, the PackageVersion will generally be int
 
 If any letters, spaces, or other non-numeric characters (other than dots) are added, the PackageVersion will fallback to string interpretation. 
 
-- Digits are sorted independently, and version `3.9.E` would be sorted higher than `3.10.B`. because 9 is larger than 1. This system doesn't have the concept of version parameters. 
-- This includes all versions. So if a package has manifests with PackageVersions `0.1.1`, `v0.1.3`, and `v0.1.4` in the winget-pkgs repo, then the string sorting determines `0.1.1` as the highest version number. 
+- Winget breaks at each dot and compares each part individually. Where a part contains an numeral-alpha, the numeral is considered before the alpha.
```

Comment:
😄 I didn't mean that this should be included literally, I was hoping you might be able to take a look at how the versions are actually sorted and update the markdown to match

Reply:
Some parts aren't the most clear - I'm not sure about the `other Other`. (Converted to Polyglot)

- If Part is less than the other Part
  - if Integer is less than other Integer, return true.
    - else if Integer is greater than other Integer, return false.
    - else if Other is empty, return false. (If this Other is empty, it is at least is greater than or equal to.)
    - else if Other is not empty and other Other is empty, return true. (If the other Other is empty and this is not, this is less.)
    - else if m_foldedOther is less than other m_foldedOther, return true. (Compare the folded versions.)
	(else Other is greater than or equal to other Other, return false.)
- If Part is equal to other Part, return Integer is equal to the other Integer and m_foldedOther is equal to other m_foldedOther.
- If Part is not equal to other Part, return pointer-this is not equal to other 
- If Part is greater than other Part, return m_channel is less than other m_channel.

Reply:
`other Other` is a bit confusing for sure!

Consider the version `1.2.34ab5`. Breaking it into parts looks like this, where each line is a `Part`. To compare a version, you start at the leftmost `Part`, if they are equal then you continue to the next `Part`. If they are not equal, the `Part` which is greater belongs to the `Version` which is greater

| Integer | Other |
| --- | --- |
| 1 | null |
| 2 | null |
| 34 | ab5 |

So when comparing `1.2.34ab5` and `1.2.34`
| this.Integer | this.Other | other.Integer | other.Other |
| --- | --- | -- | -- |
| 1 | null | 1 | null |
| 2 | null | 2 | null |
| 34 | ab5 | 34 | null |

1. Is `Integer` less than `other Integer`? 
  a. No, they are equal, continue
2. Is `other` present ? 
  a. No, it is not. Since `this other` is empty, the version is at least greater than or equal to, return false

\- Next Part -
1. Is `Integer` less than `other Integer`? 
  a. No, they are equal, continue
2. Is `other` present ? 
  a. No, it is not. Since `this other` is empty, the version is at least greater than or equal to, return false

\ - Next Part -
1. Is `Integer` less than `other Integer`? 
  a. No, they are equal, continue
2. Is `other` present ? 
  a. Yes, it is, continue
3. Is `other Other` present?
  b. No, it is not. `this` Part is less than `other`, return true.
  (Note: this seems counter-intuitive that it would be less, but is what causes 1.2.34 to be considered greater than 1.2.34-rc or `1.2.34b`)